contracts/debtAllocators/GenericDebtAllocator.sol

```diff
@@ -0,0 +1,211 @@
+
+    // Address of the vault this serves as allocator for.
+    address public vault;
+
+    // Total debt ratio currently allocated in basis points.
+    // Can't be more than 10_000.
+    uint256 public debtRatio;
+
+    // Max the chains base fee can be during debt update.
+    // Will default to max uint256 and need to be set to be used.
+    uint256 public maxAcceptableBaseFee;
+
+    constructor(address _vault, address _governance) Governance(_governance) {
+        inizialize(_vault, _governance);
+    }
+
+    /**
+     * @notice Initializes the debt allocator.
+     * @dev Should be called atomiclly after cloning.
+     * @param _vault Address of the vault this allocates debt for.
+     * @param _governance Address to govern this contracts.
+     */
+    function inizialize(address _vault, address _governance) public {
+        require(address(vault) == address(0), "!initialized");
+        vault = _vault;
+        governance = _governance;
+        // Default max base fee to uint256 max
+        maxAcceptableBaseFee = type(uint256).max;
+    }
+
+    /**
+     * @notice Check if a strategies debt should be updated.
+     * @dev This should be called by a keeper to decide if a strategies
+     * debt should be updated and if so by how much.
+     *
+     * This cannot be used to withdraw down to 0 debt.
+     *
+     * @param _strategy Address of the strategy to check.
+     * @return . bool repersenting if the debt should be updated.
+     * @return . calldata that includes what the debt should be updated to.
+     */
+    function shouldUpdateDebt(
+        address _strategy
+    ) external view returns (bool, bytes memory) {
+        // Check the base fee isn't to high.
+        if (block.basefee > maxAcceptableBaseFee) {
+            return (false, bytes("Base Fee"));
+        }
+
+        // Cache the vault variable.
+        IVault _vault = IVault(vault);
+        // Retrieve the strategy specifc parameters.
+        IVault.StrategyParams memory params = _vault.strategies(_strategy);
+        // Make sure its an active strategy.
+        require(params.activation != 0, "!active");
+
+        // Get the strategy specific debt config.
+        Config memory config = configs[_strategy];
+        // Make sure we have a target debt.
+        require(config.targetRatio != 0, "!target");
+
+        // Get the target debt for the strategy based on vault assets.
+        uint256 targetDebt = Math.min(
+            (_vault.totalAssets() * config.targetRatio) / MAX_BPS,
+            // Make sure it is not more than the max allowed.
+            params.max_debt
+        );
+
+        // If we need to add more.
+        if (targetDebt > params.current_debt) {
+            // We can't add more than the current idle.
+            uint256 toAdd = Math.min(
+                targetDebt - params.current_debt,
+                // This may underflow if idle is less than the min.
+                // Thats okay since it should return false anyway.
```

Comment:
Maybe we should we be consistent on whether we revert or return false, like we do in some cases above. what do you think?

Reply:
Yeah I think your right. We will be happy to have specific reasons for each scenario.

https://github.com/yearn/vault-periphery/pull/15/commits/2da4e61230b1c789a66f285e62d5144c2d587d98

I would like to keep the require statements for the activation and targetRatio. Because those are things that require intervention in order to fix and wont change without it.

I did just put a general "Below Min" at the bottom to avoid extra else {} sections that cluttered it. Not sure if we would get a benefit of the difference between specific reasons for not being above the min to add, pull or being at the exact right amount.